src/components/analytics/MembersChart/MembersChart.tsx

```diff
@@ -0,0 +1,205 @@
+import {
+  Box,
+  Center,
+  HStack,
+  Select,
+  Spinner,
+  Text,
+  Tooltip,
+  useBreakpointValue,
+} from "@chakra-ui/react"
+import { Axis, Grid, LineSeries, XYChart, allColors } from "@visx/xychart"
+import useGuild from "components/[guild]/hooks/useGuild"
+import Card from "components/common/Card"
+import ErrorAlert from "components/common/ErrorAlert"
+import useSWRWithOptionalAuth from "hooks/useSWRWithOptionalAuth"
+import { useEffect, useMemo, useState } from "react"
+import { Role } from "types"
+import getRandomInt from "utils/getRandomInt"
+import MembersChartLinesPanel from "./components/MembersChartLinesPanel"
+import MembersChartTooltip from "./components/MembersChartTooltip"
+
+export type MemberCountData = {
+  count: number
+  timestamp: string
+}
+
+export type MembersChartAccessors = {
+  xAccessor: (d: MemberCountData) => any
+  yAccessor: (d: MemberCountData) => any
+}
+
+type MemberCountByRole = { roleId: number; memberCounts: MemberCountData[] }
+
+const accessors: MembersChartAccessors = {
+  xAccessor: (d) => new Date(d?.timestamp),
+  yAccessor: (d) => d.count,
+}
+
+const CHART_MARGIN_X = 18
+
+const date = new Date()
+const current = date.toISOString()
+// we only have correct data since 26 october
+const startDate = new Date("2023-10-26")
+const prev = startDate.toISOString()
+
+const MembersChart = () => {
+  const { id, roles } = useGuild()
+  const chartHeight = useBreakpointValue({ base: 250, md: 500 })
+  const numberOfDateMarkers = useBreakpointValue({ base: 2, sm: 4 })
+
+  const { data, isLoading, error } = useSWRWithOptionalAuth<{
+    roles: MemberCountByRole[]
+    total: MemberCountData[]
+  }>(
+    id
+      ? `/v2/analytics/guilds/${id}/member-counts?fetchTotal=true&from=${prev}&to=${current}`
+      : null
+  )
+
+  const sortedRoles: Role[] = useMemo(() => {
+    const byMembers = roles?.sort(
+      (role1, role2) => role2.memberCount - role1.memberCount
+    )
+    return byMembers
+  }, [roles])
+
+  const [shownLines, setShownLines] = useState(["total"])
+  useEffect(() => {
+    if (!roles) return
+    setShownLines(["total", ...roles?.map((role) => role.id.toString())])
+  }, [roles])
+
+  const roleColors: Record<number, string> = useMemo(
+    () =>
+      roles?.reduce((acc, curr) => {
+        acc[curr.id] =
+          /* curr.imageUrl
+          ? await getColorByImage(curr.imageUrl)
+          : */ Object.values(allColors)[getRandomInt(9)][4]
```

Comment:
I think this should be `getRandomInt(12)`, right? There are 13 different colors inside `allColors`, and each color array has 9 hex codes inside it (I guess that's why you selected the 4th index) 👀